asab/metrics/influxdb.py

```diff
@@ -153,27 +153,35 @@ def get_field(fk, fv):
 	return field
 
 
-def combine_tags_and_field(tags, values):
+def combine_tags_and_field(tags, values, timestamp):
 	# First escape tags and values
 	tags = escape_tags(tags)
 	values = escape_values(values)
 	# Then combine the tags and then values
 	tags_string = ",".join(["{}={}".format(tk, tv) for tk, tv in tags.items()])
 	field_set = ",".join([get_field(value_name, value) for value_name, value in values.items()])
-	return tags_string + " " + field_set
+	return tags_string + " " + field_set + " " + str(int(timestamp * 1e9))
 
 
-def build_metric_line(tags, values, upperbound=None):
+def build_metric_line(tags, values, timestamp, upperbound=None):
 	if upperbound is not None:
 		tags["le"] = upperbound
-	return combine_tags_and_field(tags, values)
+	return combine_tags_and_field(tags, values, timestamp)
 
 
-def metric_to_influxdb(metric_record, now):
-	if metric_record.get("@timestamp") is None:
+def get_timestamp(metric_record, field, now):
+	fieldset = metric_record.get("fieldset")
+	if fieldset is None or len(fieldset) == 0:
 		timestamp = now
 	else:
-		timestamp = metric_record.get("@timestamp")
+		if "measured@" in field:
+			timestamp = field["measured@"]
+		else:
+			timestamp = now
```

Comment:
As I understand it now... "measured@" should always be in the field, right? If it is not, something is wrong...?